solr/core/src/java/org/apache/solr/schema/StrField.java

```diff
@@ -68,6 +71,20 @@ public List<IndexableField> createFields(SchemaField field, Object value) {
     return Collections.singletonList(fval);
   }
 
+  @Override
+  public Query getPrefixQuery(QParser parser, SchemaField sf, String termStr) {
+    final var query = super.getPrefixQuery(parser, sf, termStr);
+
+    // Some internal usage (e.g. faceting) creates PrefixQueries without a surrounding QParser, so
+    // check for null here before using QParser to access the limit value
+    if (query instanceof PrefixQuery && parser != null) {
+      final var minPrefixLength =
+          parser.getReq().getCore().getSolrConfig().prefixQueryMinPrefixLength;
+      QueryUtils.ensurePrefixQueryObeysMinimumPrefixLength(query, termStr, minPrefixLength);
```

Comment:
At this juncture, I think you should call parser.getPrefixQueryMinPrefixLength so that a QParser has the opportunity to choose a default.  The default implementation of this method can be as you indicate.